src/multiom/encoders/encoder-grib2/grib2-section4/grib2-ensemble-configurator/grib2_section4_ensemble_default_mod.F90

```diff
@@ -677,7 +682,24 @@ PP_THREAD_SAFE FUNCTION GRIB2_SECTION4_ENSEMBLE_DEFAULT_PRESET( THIS, &
 
   !> Configure the information needed for describing the ensemble (tipically 1 from table 4.6:
   !> Unperturbed low-resolution control forecast )
-  PP_METADATA_SET( METADATA, ERRFLAG_METADATA, 'typeOfEnsembleForecast', PAR%ENSEMBLE%TYPE_OF_ENSEMBLE_FORECAST_ )
+  IF ( PAR%ENSEMBLE%TYPE_OF_ENSEMBLE_FORECAST_ .NE. UNDEF_PARAM_E ) THEN
+    PP_METADATA_SET( METADATA, ERRFLAG_METADATA, 'typeOfEnsembleForecast', PAR%ENSEMBLE%TYPE_OF_ENSEMBLE_FORECAST_ )
+  ELSE
+    IF ( MSG%TYPE .NE. UNDEF_PARAM_E ) THEN ! Has( number )
```

Comment:
the comment and the error `ERRFLAG_NUMBER_EXPECTED` are not correct?? we are checking MSG%TYPE ?